### ISSUE TITLE
Plan client side files.

### ISSUE THREAD

**Issue by MichaelPachec0** [MichaelPachec0/nodejs-project-night#6]
This will be used as conversation on how the client will be structured and will reference other issues pertaining to more specifics. This should be assigned to everyone in the group as this will be a group effort. The specific issues that be referenced here will talk about what the files will do (html/js/css) can be specifically assigned.

Issues to tackle:

**Comment by MichaelPachec0** [comment]
For now there isnt any issues but as people comment here issues can be opened and tracked.

**Comment by JoeDravarol** [comment]
After re-reading the conversations I think we are using this [Frontend Mentor](https://www.frontendmentor.io/challenges/memory-game-vse4WFPvM) as design inspiration provided by [@istarlet](https://github.com/istarlet). [Alysanne](https://github.com/jericashall) mention that we could use pokemon cards instead for the memory game part. 

I think within the time scope we will only be able to tackle the main memory game feature. We would need someone to find some pokemon cards image first, ideally less than 3-6 would be enough since we would have so setup the request for the images on each request in the backend.

**Comment by JoeDravarol** [comment]
These was some issues suggestion provided by Alysanne to tackle on the Frontend: 

Issues to tackle:
- Setting up fetch request to the backend.
- Displaying cards in the DOM.
- Flipping animations
- Logic to identify match cards.
- Creating the layout.

**Comment by JoeDravarol** [comment]
I think it would be great to assign people to tackle #15 and #16 today, so that we have the base styling ready for when we will tackle the DOM logics on Monday.

**Comment by MichaelPachec0** [comment]
> I think it would be great to assign people to tackle #15 and #16 today, so that we have the base styling ready for when we will tackle the DOM logics on Monday. 

Sure, just in case be sure to post it in the discord channel so everyone knows about it.

**Comment by harris2310** [comment]
Hello Michael,
I'll work on issue #15 and see what I can do.

**Comment by JoeDravarol** [comment]
> > I think it would be great to assign people to tackle #15 and #16 today, so that we have the base styling ready for when we will tackle the DOM logics on Monday.
> 
> I agree, just in case be sure to post it in the discord channel so everyone knows about it.

I've asked @jericashall for permission just waiting for approval. I'm currently at work at the moment, once I have an approval to start #15 #16 I'll let you know and it would be great if you could post it on Discord on my behalf. Thanks in Advance!

**Comment by JoeDravarol** [comment]
> 



> Hello Michael, I'll work on issue #15 and see what I can do. Should it be vanilla Javascript, or can I use react for it.

Ideally it should be in vanilla JavaScript because majority of the member has not learn React and this project is more targeted to beginners. It would be great if you don't start on the issues just yet since we are waiting for @jericashall approval and that we can have multiple colloborate on it after the Office hours.

**Comment by tmills0203** [comment]
> > 
> 
> > Hello Michael, I'll work on issue #15 and see what I can do. Should it be vanilla Javascript, or can I use react for it.
> 
> **Note**: #15 should just be HTML and CSS only. We'll be working on the JavaScript DOM and NodeJs on Monday I believe.
> 
> Ideally it should be in vanilla JavaScript because majority of the member has not learn React and this project should be beginner friendly. It would be great if you don't start on the issue just yet since we are waiting for @jericashall approval so that we can have multiple people colloborate on it after the Office hours today.

I agree that vanilla JS should be used since we haven't learn react yet.

**Comment by harris2310** [comment]
Okay I'll start working on it,

Thanks for the communication!

**Comment by tmills0203** [comment]
Don't worry about starting yet, I think the others would like to discuss the design first.

**Comment by harris2310** [comment]
Sure, let me know if there's something I can start with right now!

**Comment by tmills0203** [comment]
> Sure, let me know if there's something I can start with right now!

hey Harris, I got the ok from @jericashall to go ahead start on #15.

**Comment by jericashall** [comment]
I responded to a couple people on discord but wanted to write here, too. 
For issue #15 - I think people are interpreting this in different ways. Harris seems to be interpreting it as the javascript logic for the game, and Joe is interpreting it as page layout structure with css.
I think we definitely want someone to work on html/css layout ahead of time. Whoever wants to take that issue is fine. Someone could also outline the game logic but getting the specifics done will be hard without the api finished.
Issue #16 should be fine to work on, you can pick if you'd like to work with JS or CSS animations.

As for react - I don't think we should use that on this project. Lots of the project contributors are only familiar with vanilla javascript at this point.

Also, Joe mentioned someone finding Pokemon cards to use. Someone in the API issue said they had an idea for that, so I'm waiting to hear back from them.

**Comment by JoeDravarol** [comment]
> I responded to a couple people on discord but wanted to write here, too. For issue #15 - I think people are interpreting this in different ways. Harris seems to be interpreting it as the javascript logic for the game, and Joe is interpreting it as page layout structure with css. I think we definitely want someone to work on html/css layout ahead of time. Whoever wants to take that issue is fine. Someone could also outline the game logic but getting the specifics done will be hard without the api finished.

In regards to issue #15 I think it should only be page layout structure with css because we don't have any data to work with yet as it will be coming from the back end. Also since there are 13 of us collaborating, I am thinking we should split each features into its own issue so everyone will be able to contribute at least once. So If @harris2310 was to work on both the layout stucture, CSS and JavaScript a lot of people will not have the opportunity to able to contribute. 

Unless we assign multiple people on the same issue, where the assign collaborators are in a Discord voice chat; we could have one person sharing their screen as the main coder and bounce idea of each other. I have seen people doing it this way during the group project night. This is open for discussion.

**Comment by jericashall** [comment]
Yeah, I agree the JS will be hard to do without the API. We can split out issues into smaller or tickets or have multiple people work on one ticket. We're trying to plan a synchronous time on Monday for most people to meet, so we could have some people work on things together during that time.

**Comment by MichaelPachec0** [comment]
as i mentioned to the group in the vc chat. As front end devs you are in charge of what output you get, you get to decide for the most part what the output should be. Assume that back-end developers will give you everything you need to start the project and work from there. 

The mockup in #15 is a great example of what to expect, you have a general idea that you expect an array of strings that contain a poke image. We also know they [array] needs to be randomized, the back-end also needs to take of that since that was what was specified as a requirement as part of the project.

You also need a way of keeping score. This can be setup either as sending a separate object, or a big meta object that contains everything.
 This is a mockup object that could be sent (i opted to send it as one big object because its going to be easier for you and the backend groups (both know what data is expected to be sent to the client):
```js
resp = {
    name: "name of the player",
    points: 100,
    level: 3,
   //This might be a little too advanced but since the mockup defines it then
   // im including it here
    timer: date_object
   // array of links to pokemon in question
   // since this is a simple game worrying about cheaters should be secondary issue, 
   // more important is get this working 
   poke_array: []
}

```
take this idea and make it you own front end 💯 devs

**Comment by harris2310** [comment]
Got it, I'll just write the css and html structure of the tiles

**Comment by MichaelPachec0** [comment]
just posted the spec in the back-end api issue as a comment [here](https://github.com/MichaelPachec0/nodejs-project-night/issues/4#issuecomment-1144533874)
any questions welcome here or in the issue page itself
